test/TestCaseReader.cpp

```diff
@@ -145,8 +146,9 @@ pair<map<string, string>, size_t> TestCaseReader::parseSourcesAndSettingsWithLin
 		else
 			throw runtime_error(string("Expected \"//\" or \"// ---\" to terminate settings and source."));
 	}
+	// Register last source
 	sources[currentSourceName] = currentSource;
-	return { sources, lineNumber };
+	return { { sources, currentSourceName }, lineNumber };
```

Comment:
```suggestion
	return {{sources, currentSourceName}, lineNumber};
```
I generally only see this style used for function braces in the codebase. The coding style doesn't say anything about it (only not to add spaces inside parentheses) - maybe we should update it to be clear about `[]` and `{}` too?

Also, I think you need an explicit `move` here (compiler won't do it automatically):
```suggestion
	return {{move(sources), move(currentSourceName)}, lineNumber};
```